### PR TITLE
Refactor map components and introduce zones

### PR DESCRIPTION
**Pull request by hsablonniere**

### CODE REVIEW

File: src/maps/cc-logsmap.js [hunk 1]
```diff
@@ -83,15 +86,84 @@ export class CcLogsMap extends LitElement {
...
+    // Add point now
+    if (this._pointsByCoords[coords] == null) {
+      this._pointsByCoords[coords] = [];
+    }
+    this._pointsByCoords[coords].push(newPoint);
+
+    this._updatePoints();
+
+    // Schedule delete point
+    setTimeout(() => {
```
Comment (BlackYoup) [inline src/maps/cc-logsmap.js:125]: Don't you want to abort the timeout when the component is "dropped"?

File: src/zones/cc-zone-input.js [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+ * A input component to select a zone with a map and a list.
+ *
+ * * 🎨 default CSS display: `grid`
+ * <br>
+ * 🧐 [component's source code on GitHub](https://github.com/CleverCloud/clever-components/blob/master/src/zones/cc-zone-input.js)
+ *
+ * ## Details
+ *
+ * * When `zones` is nullish, a skeleton screen UI pattern is displayed (loading hint).
+ * * Zones are sorted in the list using `tags`. Clever Cloud, then private, the alphanum sort on the city name.
```
Comment (BlackYoup) [inline src/zones/cc-zone-input.js:25]: did you mean `then alphanum`?

File: src/zones/cc-zone-input.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+      this.shadowRoot.querySelector(`input[id=${name}]`).scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest',
+      });
+    }, 200);
+  }
+
+  _renderZoneInput (zone) {
+
+    // Simplified template withou label/input when skeleton is enabled
```
Comment (BlackYoup) [inline src/zones/cc-zone-input.js:198]:
```suggestion
    // Simplified template without label/input when skeleton is enabled
```

File: src/maps/cc-map.js [hunk 4]
```diff
@@ -223,109 +201,128 @@ export class CcMap extends withResizeObserver(LitElement) {
...
-
-    // Compute total count of the points at those coordinates
-    // We don't use this value for the tooltip for now
-    const totalCount = this._points
-      .filter((p) => p.coords === point.coords)
-      .map((p) => p.count)
-      .reduce((a, b) => a + b, 0);
+    const obsoleteIds = new Set(Object.keys(this._pointsCache));
+
+    // Create new markers or update them if there's already a one with the same tag at the same coordinates
```
Comment (BlackYoup) [inline src/maps/cc-map.js:212]:
```suggestion
    // Create new markers or update them if there's already one with the same tag at the same coordinates
```

File: src/zones/cc-zone-input.js (hunk 2 above)
Comment (hsablonniere) [reply src/zones/cc-zone-input.js:25]: I meant "classic alphanumeric sort". I fixed that and also replaced the "the" with "then".

File: src/maps/cc-logsmap.js (hunk 1 above)
Comment (hsablonniere) [reply src/maps/cc-logsmap.js:125]: You mean if the component is removed from the DOM?
It's the code we had in `<cc-map>`, as is, I just move it.
Maybe we could investigate this later...

https://github.com/CleverCloud/clever-components/issues/158

**Comment by hsablonniere** [comment]
@BlackYoup Thanks for the review :+1:

File: src/maps/cc-logsmap.js (hunk 1 above)
Comment (BlackYoup) [reply src/maps/cc-logsmap.js:125]: > You mean if the component is removed from the DOM?

Yes. I don't like having timeouts that won't do anything anymore running in the background. Sure, let's check it later :+1:

**Pull request merged**